File: Site de desafios/beecrowd/menordapilha.java
import java.util.*;

class Pilha{
    int tam;
    int []pilha;
    int topo;

    public Pilha(int t){
        this.tam = t;
        this.pilha = new int[tam];
        this.topo = -1;
    }

    public boolean inserirpilha(int valor){
        boolean flag = false;
        if(topo > tam -1){
            System.out.print("ERRO");
            return flag;
        }
        topo++;
        pilha[topo] = valor; 
        flag = true;
        return flag;
    }

    public int removerpilha(){
         if(topo < 0){
            System.out.print("ERRO");
            return -1;
        }
        int removido = pilha[topo];
        topo--;
        return removido;
    }

    public int menorvalorpilha(){
        int menor = pilha[0];
        for(int i = 1;i<=topo;i++){
            if(menor > pilha[i]){
                menor = pilha[i];
            }
        }
        return menor;
    }

    public void imprimirpilha(){ 
        for (int i = topo ; i >= 0;i--){
            System.out.print( pilha[i] + " ");
        }
    }
}

public class menordapilha{

    public static void main(String[] Args){
        Scanner sc = new Scanner(System.in);
        int acoes = sc.nextInt();
        sc.nextLine();
        Pilha pil = new Pilha(acoes);
        while(acoes > 0 ){
            String acao = sc.next();
            if(acao.equals("PUSH")){ 
                int num = sc.nextInt();
                sc.nextLine();
                System.out.println(pil.inserirpilha(num));
            }
            else if(acao.equals("MIN")){
                System.out.println(pil.menorvalorpilha());
            }else if(acao.equals("POP")){
                pil.removerpilha();
            }
            
            acoes --;
        }
        sc.close();
    }
}

File: Site de desafios/beecrowd/MenorDaPilha.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

typedef struct Celula{
    struct Celula* prox;
    int elemento;
}Celula;

typedef struct Pilha
{
    Celula* topo;
}Pilha;

void PUSH(Pilha* pilha, int x){
    Celula* tmp = (Celula*)malloc(sizeof(Celula));
    if(tmp == NULL){
        printf("Erro de alocaçao na celula push\n");
        return;
    }
    tmp->elemento = x;
    tmp->prox = pilha->topo;
    pilha->topo = tmp;
}

int POP(Pilha* pilha){
    Celula* tmp = (Celula*)malloc(sizeof(Celula));
    if(pilha->topo == NULL){
        printf("EMPTY\n");
        return -1; 
    }
    int removido = pilha->topo->elemento;
    tmp = pilha->topo;
    pilha->topo = pilha->topo->prox;
    free(tmp);
    return removido;
}

int MIN(Pilha* pilha){
    if(pilha->topo == NULL){
        printf("EMPTY\n");
        return -1; 
    }
    int menor = pilha->topo->elemento;
    for(Celula* tmp = pilha->topo; tmp != NULL; tmp = tmp->prox){
        if(tmp->elemento < menor){
            menor = tmp->elemento;
        }
    }
    return menor;
}

int main(){
    int N=0;
    scanf("%d", &N);
    Pilha pilha;
    pilha.topo = NULL;
    for(int i=0; i<N; i++){
        char op[5];
        scanf("%4s",op);
        if(strcmp(op, "PUSH") == 0){
            int x=0;
            scanf("%d",&x);
            PUSH(&pilha,x);
        }
        if(strcmp(op, "POP") == 0){
            int x = POP(&pilha);
        }
        if(strcmp(op, "MIN") == 0){
            int x = MIN(&pilha);
              if(x != -1){
                printf("%d\n",x);
            }
        }
    }
    return 0;
}